### ISSUE TITLE
[Feature Request]: Identifier-case-sensitivity Compatibility

### ISSUE THREAD

**Issue by sukki37** [matrixorigin/matrixone#2208]
### Is there an existing issue for the same feature request?

- [X] I have checked the existing issues.

### Is your feature request related to a problem?

```markdown
MO behaves inconsistently with MySQL in terms of identifier-case sensitivity of columns.
```

### Describe the feature you'd like

For compatibility with MySQL, we'd better follow its identifier-case-sensitivity rules(https://dev.mysql.com/doc/refman/8.0/en/identifier-case-sensitivity.html).
<img width="1028" alt="image" src="https://user-images.githubusercontent.com/77312370/163085876-e05c4a12-3e12-49ad-83f3-0a14bc8ef84e.png">

**Comment by m-schen** [comment]
This is a semantic issue that needs to be discussed and decided. @yingfeng @JinHai-CN @nnsgmsone

**Comment by iamlinjunhong** [comment]
0.5 confirms that identifiers are not case sensitive. Full compatibility with MySQL requires discussion and investigation

**Comment by sukki37** [comment]
closed, tracking in #2454
